### DISCUSSION TITLE
SiliconGraphics Indy support / Irix

### DISCUSSION THREAD

**Discussion by Sch-LikA** [erichelgeson/BlueSCSI-v1#107] [Q&A]
I just tried setting up a SGI Indy on a 20220404 based 1.1b board I just assembled. With just one file on the exfat formatted sd, HD10_512IndyIrix53.hda, at boot I get a bad error :

sc0,1,0: cmd=0x12 Unexpected info phase 47, state 40. Resetting SCSI bus

this message loops for a while and the PROM Diagnostics fails.

A scsi2sd seems to working ok.

Is there anything I can do to provide more log/debug ?

Btw, I have quite a few "exotic" scsi machines, from HP, Sun, SGI, Commodore, Atari, NeXT, Apple or Roland, I may come with some other test results over time, don't know if you'd like me to test some in a certain order, just tell me ! I have also some coding knowledge but could not find any scsi protocol documentation in a digestable format, do you have any pointer ? I may help.

Anyway thanks a lot for your work, I really appreciate it.

**Comment by gammy** [comment]
I just tried the BlueSCSI (HW rev `1.1-a`, FW rev `1.1-20220404-USB`) and am getting a similar (but not identical) issue on boot on my Indy.
For me, the following message loops forever:
```
WARNING: wd93 SCSI Bus=0 ID=5 LUN=0: SCSI cmd=0x1a
Unexpected info phase 36, state 49. Resetting SCSI bus
```
See also [this photo](https://gammy.ere.nu/BlueSCSI/SGI/220526-Indy-unexpected_info-reset.jpg).
The ID and LUN are correct for my image.

As you also mention Sun, see also the [Anyone tried BlueSCSI with a Sun Sparcstation?](https://github.com/erichelgeson/BlueSCSI/discussions/50) discussion - where BlueSCSI also does not, as of writing, work properly - which also mentions further debugging suggestions.

**Reply by erichelgeson** [reply]
cmd 0x1a is sense 6 - which in `main` i've added the flexible geometry info to just recently, not sure if thats what it's looking for but if you wanted to try flashing main, or wait till this weekend for me to do the release, it may help. It would be good if it told us what was unexpected about the response :)

**Comment by erichelgeson** [comment]
cmd 0x12 is an Inquiry, which we respond with just a 36 byte array - https://github.com/erichelgeson/BlueSCSI/blob/main/src/BlueSCSI.cpp#L286 - if we knew what it was looking for we could add it.

**Reply by jgilje** [reply]
I've also got a quite similar setup, an Indy + bluescsi with the inquiry commands failing. I turned on debug logging to the SD card if that helps

```
BlueSCSI <-> SD - https://github.com/erichelgeson/BlueSCSI
VERSION: 1.1-SNAPSHOT-20220522
DEBUG:2 SDFAT_FILE_TYPE:3
SdFat version: 2.1.2
Sd Format: exFAT
SPI speed: 50Mhz
SdFat Max FileName Length: 32
Initialized SD Card - lets go!
Sd MID:3 OID:SD
Sd Name:SC32G
Sd Date:8/2011
Sd Serial:1917511629
 - HD10_512.hda / 268435456bytes / 262144KiB / 256MiB
ID:LUN0:LUN1:
 0:----:----:
 1: 512:----:
 2:----:----:
 3:----:----:
 4:----:----:
 5:----:----:
 6:----:----:
Finished initialization of SCSI Devices - Entering main loop.
BusReset!
Selection
Command:BusReset!
Reset, going to BusFree
BusFree
Selection
Command:BusReset!
Reset, going to BusFree
BusFree
Selection
Command:BusReset!
Reset, going to BusFree
BusFree
Selection
Command:BusReset!
Reset, going to BusFree
BusFree
Selection
Command:BusReset!
Reset, going to BusFree
BusFree
Selection
Command:BusReset!
```

Is that of any help?
